common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/Work.scala

```diff
@@ -113,21 +112,55 @@ sealed trait WorkState {
 
 object WorkState {
 
-  // TODO: for now just 2 states, in the end the states will correspond to the
-  // block comment above
-
-  case class Unidentified(
+  case class Source(
     sourceIdentifier: SourceIdentifier
   ) extends WorkState {
 
     type WorkDataState = DataState.Unidentified
   }
 
+  case class Merged(
+    sourceIdentifier: SourceIdentifier,
+    isMerged: Boolean
+  ) extends WorkState {
+
+    type WorkDataState = DataState.Unidentified
+  }
+
+  case class Denormalised(
+    sourceIdentifier: SourceIdentifier,
+    isMerged: Boolean = false
+  ) extends WorkState {
+
+    type WorkDataState = DataState.Unidentified
+  }
+
   case class Identified(
     sourceIdentifier: SourceIdentifier,
     canonicalId: String,
+    isMerged: Boolean = false
   ) extends WorkState {
 
     type WorkDataState = DataState.Identified
   }
 }
+
+object WorkFsm {
+
+  import WorkState._
+
+  implicit class TransitionSourceWork(work: Work[Source]) {
+
+    def transitionToMerged(isMerged: Boolean): Work[Merged] = {
```

Comment:
Not sure about the naming here - `transitionToMerged(isMerged = false)` seems sort of self-contradictory. Now I think about it, perhaps the naming problem is really with the `isMerged` field (the contradiction is in `Merged(isMerged = false)`)... Some suggestions I haven't thought about much:

- `pure`
- `hasMultipleSources`
- `passedThrough`
- `singleSource` (negated)

Reply:
I think `hasMultipleSources` is really clear. @jamesgorrie any thoughts?

Reply:
Why not just `toMerged` and `hasMultipleSources`? Could that be derived from the `otherIdentifiers` property?

Reply:
The `merged` field is never actually used as far as I can tell, I think it would probably be useful sometimes for filtering things in the elasticsearch index, in which case we need an explicit field rather than a derived one